[Input plugin] - [gNMI - tag_only] Assert string fails and crashes collection

Regarding: https://github.com/influxdata/telegraf/blob/fc8301aeedef9110a21dfdce301428c79d6f45c5/plugins/inputs/gnmi/gnmi.go#L329

Telegraf crashes with similar errors when assert string fails:
```
2022-04-20T15:13:59Z I! Using config file: /etc/telegraf/telegraf.conf
2022-04-20T15:13:59Z I! Starting Telegraf 1.22.1
2022-04-20T15:13:59Z I! Loaded inputs: gnmi
2022-04-20T15:13:59Z I! Loaded aggregators:
2022-04-20T15:13:59Z I! Loaded processors:
2022-04-20T15:13:59Z I! Loaded outputs: influxdb
2022-04-20T15:13:59Z I! Tags enabled: host=90365323743d
2022-04-20T15:13:59Z I! [agent] Config: Interval:10s, Quiet:false, Hostname:"90365323743d", Flush Interval:10s
2022-04-20T15:13:59Z D! [agent] Initializing plugins
2022-04-20T15:13:59Z D! [agent] Connecting outputs
2022-04-20T15:13:59Z D! [agent] Attempting connection to [outputs.influxdb]
2022-04-20T15:13:59Z D! [agent] Successfully connected to outputs.influxdb
2022-04-20T15:13:59Z D! [agent] Starting service inputs
2022-04-20T15:13:59Z D! [inputs.gnmi] Connection to gNMI device A.B.C.D:55555 established
2022-04-20T15:13:59Z D! [inputs.gnmi] Connection to gNMI device A.B.C.D:55555 closed
panic: interface conversion: interface {} is uint64, not string

goroutine 16 [running]:
github.com/influxdata/telegraf/plugins/inputs/gnmi.(*GNMI).handleSubscribeResponseUpdate(0xc00041b380, {0xc000a70ea1, 0x12}, 0xc000130b58)
	/go/src/github.com/influxdata/telegraf/plugins/inputs/gnmi/gnmi.go:329 +0xf79
github.com/influxdata/telegraf/plugins/inputs/gnmi.(*GNMI).handleSubscribeResponse(0xc00052e670, {0xc000a70ea1, 0xc00046a510}, 0xc000071720)
	/go/src/github.com/influxdata/telegraf/plugins/inputs/gnmi/gnmi.go:264 +0x4e
github.com/influxdata/telegraf/plugins/inputs/gnmi.(*GNMI).subscribeGNMI(0xc00041b380, {0x5e68590, 0xc00046a510}, {0xc000a70ea1, 0x12}, 0x0, 0x0)
[github_test_npu_gnmi_tags.conf.txt](https://github.com/influxdata/telegraf/files/8522927/github_test_npu_gnmi_tags.conf.txt)

	/go/src/github.com/influxdata/telegraf/plugins/inputs/gnmi/gnmi.go:256 +0x536
github.com/influxdata/telegraf/plugins/inputs/gnmi.(*GNMI).Start.func1({0xc000a70ea1, 0x12})
	/go/src/github.com/influxdata/telegraf/plugins/inputs/gnmi/gnmi.go:158 +0x125
created by github.com/influxdata/telegraf/plugins/inputs/gnmi.(*GNMI).Start
	/go/src/github.com/influxdata/telegraf/plugins/inputs/gnmi/gnmi.go:155 +0x9d3
```

Telegraf configuration:
[github_test_npu_gnmi_tags.conf.txt](https://github.com/influxdata/telegraf/files/8522929/github_test_npu_gnmi_tags.conf.txt)


@fcuiller

--- a/plugins/inputs/gnmi/gnmi.go
+++ b/plugins/inputs/gnmi/gnmi.go
@@ -326,7 +326,7 @@ func (c *GNMI) handleSubscribeResponseUpdate(address string, response *gnmiLib.S
 		for subscriptionName, values := range c.lookup {
 			if annotations, ok := values[subscriptionKey]; ok {
 				for k, v := range annotations {
-					tags[subscriptionName+"/"+k] = v.(string)
+					tags[subscriptionName+"/"+k] = fmt.Sprint(v)
 				}
 			}
 		}

